pkg/apis/shared_types.go

```diff
@@ -0,0 +1,100 @@
+package apis
```

Comment:
Is there a reason this isn't put into the utils package or something similar? This is not a very descriptive name IMO

Reply:
Simply followed the ticket, I can move it though. What do you think? @rogbas @jharrington22

Reply:
The tickets specifies the pkg `v1alpha1` as I read it.

Reply:
https://issues.redhat.com/browse/OSD-2835 @fahlmant

Reply:
Yes apis isn't quite right. I wanted it to be part of the v1alpha1 package I just got the path wrong in the ticket. @ArielLima please update.